tests/integration.rs

```diff
@@ -345,3 +367,102 @@ fn sparse3(#[case] harness: Harness) {
 	let expected = [b'x'; 32768];
 	assert_eq!(buf, expected);
 }
+
+#[apply(all_images)]
+fn listxattr(#[case] harness: Harness) {
+	let d = &harness.d;
+
+	let file = File::open(d.path().join("xattrs")).unwrap();
+	let xattrs = file.list_xattr().unwrap().collect::<Vec<_>>();
+	let expected = [OsStr::new("user.test")];
+	assert_eq!(xattrs, expected);
+
+	freebsd! {
+		let num = unsafe { libc::extattr_list_fd(file.as_raw_fd(), EXTATTR_NAMESPACE_USER, std::ptr::null_mut(), 0) };
+		assert_eq!(num, 5); // strlen("test\0")
+	}
```

Comment:
Since this doesn't depend on the stuff up above, it should have its own test case.